Automattic/wp-calypso
# Stats: Add new commercial purchase page

Related to #80259

## Proposed Changes

* introduce a new purchase page for a single plan (not a wizard like previously)
* combine the single-page design with the existing commercial plan purchase flow
* create a new file with shared components
* extended pages visited by owners of a free or PWYW plans with a notice above the wizard

## Testing Instructions



* navigate to the live branch
* without any feature flag applied verify that the experience didn't change for:
   * a page without any plan purchased (you should see the old wizard)
   * a page with the commercial plan purchased (you should see a confirmation notice)
   * a page with free or PWYW plan purchased (you should see the old wizard)
* apply `stats/type-detection` feature flag and check the updated experience
  * if the page wasn't classified as either personal or commercial, you should see the old wizard
  * if your page has been classified as commercial you should see a new purchase page for only the commercial plan (not the wizard) - this can be tested locally by forcing commercial detection in the controller (`isCommercial` option)
  * if your page has been classified as personal you should see the wizard (a separate page is currently in progress)
  * if you purchased free or PWYW plans you should see a banner on top the wizard when visiting the page again

New commercial page:
<img width="1059" alt="SCR-20230831-nfhc" src="https://github.com/Automattic/wp-calypso/assets/112354940/67e455a1-e82d-48c1-8d2c-f6eb541d4012">

Additional pages available when visiting the purchase page with a free or personal plans:
<img width="1187" alt="SCR-20230901-obxg" src="https://github.com/Automattic/wp-calypso/assets/112354940/7d386ad8-acfb-4f6f-b847-b7c82f3f347c">

<img width="1188" alt="SCR-20230901-ocbn" src="https://github.com/Automattic/wp-calypso/assets/112354940/a23de73c-63e9-4440-84cc-d3ae3c700753">


## Pre-merge Checklist
